src/sources/linux/hardware.sh

```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env kmd
 
-exec cat /sys/devices/virtual/dmi/id/board_vendor /sys/devices/virtual/dmi/id/board_name
+exec cat /sys/devices/virtual/dmi/id/board_vendor /sys/devices/virtual/dmi/id/product_name
```

Comment:
In my experience, these `board_name` and `product_name` values vary wildly between manufacturers, and one is often unhelpful while the other is, and vice versa.

Reply:
This definitely needs improvement, as this change may satisfy your requirements, but will be unsatisfying for others.

Reply:
Good point! I agree.
I think the approach is generally too naive for the diverse Linux use case.

I am thinking of the following:
We could concatenate `board_vendor`, `product_name` and `board_name` into hardwareVersion, maybe with something like a `|` as separator.
We could then tokenize the fields in a new function that assembles the friendlyName, similar to macmodels.getFriendlyName.
That function/module could house logic for assembling the friendlyName depending on fields, like vendor, for easy extension (protected by a unit test I think).

The approach is not super clean but neither is it super intrusive.

I'll be toying around with a solution and will report back with findings when I come across them.

Any feedback welcome!

Reply:
I just pushed my take on the above.

It might become necessary though to include more fields in the concatenated hardwareVersion string.

I was just looking for some Lenovo DMI data and it seems like `/sys/devices/virtual/dmi/id/product_version` is the only data point that contains anything useful regarding which model it is. (Source: https://natalian.org/2015/02/18/Archlinux_on_a_Lenovo_X1C3/)